dipy/align/vector_fields.pyx

```diff
@@ -1866,11 +1942,71 @@ def downsample_displacement_field_2d(floating[:, :, :] field):
     return np.asarray(down)
 
 
+cdef void _warp_3d_fun(floating[:, :, :] volume, floating[:, :, :, :] d1,
+                       double[:, :] affine_idx_in,
+                       double[:, :] affine_idx_out,
+                       double[:, :] affine_disp, cnp.npy_intp k,
+                       cnp.npy_intp nrows, cnp.npy_intp ncols,
+                       floating[:, :, :] warped) nogil:
+    cdef:
+        cnp.npy_intp i, j
+        int inside
+        double dkk, dii, djj, dk, di, dj
+        floating tmp[3]
+
+    for i in range(nrows):
+        for j in range(ncols):
+            if affine_idx_in is None:
+                dkk = d1[k, i, j, 0]
```

Comment:
introducing one more for cycle instead repetitive using triple commands would make the code much more readable